### PR TITLE
Implement ProductSelect component; allow draft authors to edit their productArea

### PR DESCRIPTION
**Pull request by jeffdaley**
- Adds the ability to edit the productArea of a draft via the sidebar.
- Updates the new doc form to use the new ProductSelect component.
- Expands Mirage and test coverage

### CODE REVIEW

**Comment by jeffdaley** [comment]
> error decoding draft patch request: error="json: unknown field \"product\"

File: web/app/components/document/sidebar.js [hunk 1]
```diff
@@ -43,13 +42,28 @@ export default class DocumentSidebar extends Component {
...
+  @tracked product = this.args.document.product || "";
 
   @tracked userHasScrolled = false;
   @tracked body = null;
 
+  /**
+   * If the user is saving a new product, this will be the new product.
+   * Reset when the save task fails.
+   */
+  @tracked _newProduct = null;
```
Comment (jfreda) [inline web/app/components/document/sidebar.js:56]: Curious why we couldn't just use `this.product` for the places where we're using this?

**Review by jfreda — APPROVED** [review]
Nice, just one (non-blocking) question

File: web/app/components/document/sidebar.js (hunk 1 above)
Comment (jeffdaley) [reply web/app/components/document/sidebar.js]: Good call. Updated https://github.com/hashicorp-forge/hermes/pull/109/commits/41221b5abf09005f0e7362bd0e52ed5db1e87850

**Pull request merged**